src/eventstream.cr

```diff
@@ -0,0 +1,56 @@
+require "base64"
+
+module EventStream
+    # Each message has this structure:
+    #
+    #   total_length   : Int32  (4 bytes)
+    #   headers_length : Int32  (4 bytes)
+    #   prelude_crc    : Int32  (4 bytes)
+    #   headers        : Bytes  (headers_length bytes)
+    #   payload        : Bytes  (total_length - headers_length - 16 bytes)
+    #   message_crc    : Int32  (4 bytes)
+    #
+    # Note: We read the raw headers block here but do not parse individual header fields
+    # or verify CRC.
```

Comment:
_Should_ we parse headers and verify CRC here? I know you spent a lot of time on this, so I understand if the answer is "I have no idea, I just screamed at the top of my lungs until it parsed event payloads and I'm too tired to care", but the reason I ask is that the Crystal stdlib _might_ be able to handle both of these things for us.

- Header parsing using [`HTTP.parse_header_line`](https://github.com/crystal-lang/crystal/blob/53cead6befd1f86c061d40e659892c9c0b84b5e3/src/http/common.cr#L150), assuming it's formatted like HTTP headers
- [CRC32 checksum](https://crystal-lang.org/api/1.16.0/Digest/CRC32.html#checksum%28data%29%3AUInt32-class-method) (lacks documentation, but might just be a straight `message_crc == Digest::CRC32.checksum(payload)` comparison)

Reply:
This should work. I seem to recall the headers were just json anyway

Reply:
Made lazy, still offering the structure in the getter